app/models/session.rb

```diff
@@ -125,10 +115,28 @@ def self.from_location(data)
     where(:streams => {:id => streams_ids.uniq})
   end
 
+  def self.filter_by_time_range(sessions, time_from, time_to)
+    time_from = Time.strptime(time_from.to_s, '%s')
+    time_to = Time.strptime(time_to.to_s, '%s')
+
+    sessions.where(
+      "(start_time_local BETWEEN :time_from AND :time_to)
+      OR
+      (end_time_local BETWEEN :time_from AND :time_to)
+      OR
+      (:time_from BETWEEN start_time_local AND end_time_local)",
+      :time_from => time_from, :time_to => time_to)
+      .local_minutes_range(minutes_of_day(time_from), minutes_of_day(time_to))
```

Comment:
Would it be possible to have one query instead of two? I mean merging `filter_by_time_range` and `local_minutes_range` together?

Reply:
I'm not sure how to make it just one query since the `local_minutes_range` is only performed `unless whole_day?`.

But can I ask why you suggested that? Is it about readability? Or about performance? Cause in the latter case I think it doesn't change anything, rails take care of that. They don't execute every sql line immediately. They collect all the code that looks like sql, put it together and only execute when the resulting object is actually needed, right?

Reply:
Yeah, the query builder is lazy that's correct.

I've stared at the code for some time and I couldn't really get why there were two queries when the column is one. Thus readability is something we could work on.

The perf is something we can take a look later on I guess. But in case we wanted to use indexes here, at the moment the queries are working on
- `end_time_local`
- `start_time_local`
- `EXTRACT(HOUR FROM end_time_local)`
- `EXTRACT(HOUR FROM start_time_local)`
When a SQL function is applied to a column then the index can't be used. In other words, an index on `end_time_local` would not be used for `EXTRACT(HOUR FROM end_time_local)`